File: lc_py/lcPy/array/KLengthApart.py
from typing import List


class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        last = -1

        for i in range(n):
            if nums[i] == 1:
                if last != -1 and i - last <= k: return False
                last = i
        return True

File: lc_go/lcGo/array/KLengthApart.go
package main

func kLengthApart(nums []int, k int) bool {
	// n := len(nums)
	last := -1

	for i, x := range nums {
		if x == 1 {
			if last != -1 && i-last <= k {
				return false
			}
			last = i
		}
	}
	return true
}

func main() {

}
